CLOUDSTACK-9814 : Unable to edit a Sub domain, which has the same name in different domains

niteshsarda opened the pull request:
ISSUE
============
Unable to edit a Sub domain, which has the same name in different domains.

TROUBLESHOOTING
==================
1. Created two sub-domains with same name. Under two different domains.
ROOT

Domain 1 : Sample
AA000001
BB000001

Domain 2 : Test
AA000001

2. Try to edit resource - Click on edit - edit any parameter.

EXPECTED BEHAVIOR
==================
It should allow to edit the required parameter and update the value.

ACTUAL BEHAVIOR
==================
It throws an error "Failed to update specified domain id with name 'AA000001' since it already exists in system. 

**Screenshot before applying fix :** 

![screenshot before adding fix](https://cloud.githubusercontent.com/assets/25146827/23549581/ed9d545a-0032-11e7-8065-ec50f34ece5c.png)

**Screenshot afterapplying fix :**

![screenshot after adding fix domain 1](https://cloud.githubusercontent.com/assets/25146827/23549596/010a8148-0033-11e7-87eb-453e37031a25.png)

![screenshot after adding fix domain 2](https://cloud.githubusercontent.com/assets/25146827/23549609/0a6b8d04-0033-11e7-84b0-cdf97659679e.png)

yvsubhash commented on server/src/org/apache/cloudstack/region/RegionManagerImpl.java:
```diff
@@ -229,7 +232,7 @@ public boolean deleteUser(DeleteUserCmd cmd) {
     @Override
     public Domain updateDomain(UpdateDomainCmd cmd) {
-        return _domainMgr.updateDomain(cmd);
+        return _managementService.updateDomain(cmd);
```
Instead of changing the method that is getting called, the missing one in search query can be added here and it seems the updateDomain method in managementserviceimpl is not in use. It can be removed

niteshsarda replied:
@yvsubhash : As per your suggestion, I have changed the search query and also removed the unused method updateDomain from managementserviceimpl class.

Please check.

rafaelweingartner commented:
@niteshsarda I suggest you adding line 2281 from "managementService" to "domainMgr". The only difference between those two methods is this line. Assuming that you changed the method calls, and then everything started working. 

Then, please delete the com.cloud.server.ManagementServerImpl.updateDomain(UpdateDomainCmd), it is not used. It is incredible to have a method with ~100 duplicated between classes.

niteshsarda commented:
@rafaelweingartner : As per your suggestion, I have changed the search query and also removed the unused method updateDomain from managementserviceimpl class.

Please check.

rafaelweingartner commented:
@niteshsarda thank you very much :)
The less duplicated or useless code around the better.

The method 'updateDomain' is huge and you just changed a single line there, so I will not discuss/suggest any improvements there.

I am ok with this PR LGTM.

jayantpatil1234 commented:
I have tested this and **LGTM** for test.

niteshsarda commented:
This is Ready to Merge

cloudmonger commented:
### ACS CI BVT Run
 **Sumarry:**
 Build Number 748
 Hypervisor xenserver
 NetworkType Advanced
 Passed=108
 Failed=4
 Skipped=12

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_non_contigiousvlan.py

 * test_extendPhysicalNetworkVlan Failed

* test_volumes.py

 * test_06_download_detached_volume Failed

* test_ssvm.py

 * test_05_stop_ssvm Failed

* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failed


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

niteshsarda commented:
tag:This is Ready to Merge

Test cases which are failing are not related to my changes.
